Keen.NetStandard/Dataset/QueryDefinitionExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Keen.Core.Query;
+
+
+namespace Keen.Core.Dataset
+{
+    internal static class QueryDefinitionExtensions
+    {
+        public static void ValidateForCachedDataset(this QueryDefinition query)
+        {
+            if (string.IsNullOrWhiteSpace(query.AnalysisType))
+            {
+                throw new KeenException("QueryDefinition must have an analysis type");
+            }
+
+            if (string.IsNullOrWhiteSpace(query.EventCollection))
+            {
+                throw new KeenException("QueryDefinition must specify an event collection");
+            }
+
+            if (string.IsNullOrWhiteSpace(query.Timeframe))
+            {
+                throw new KeenException("QueryDefinition must specify a timeframe");
+            }
+
+            if (string.IsNullOrWhiteSpace(query.Interval))
```

Comment:
Is interval actually required?

Reply:
It's required by the API when creating a cached dataset. Perhaps the name of the method (`ValidateForCachedDataset`) is not clear enough.

Reply:
Yeah the docs I think aren't clear on this, which is another area for improvement.

Reply:
The docs say this:
`Currently Cached Dataset query definitions require Relative Timeframes and the unit must match the interval, `
...and this:
`Currently supports Relative Timeframes. Interval should match; eg. this_90_day and daily`
...which implies the interval should be there, but could be clearer.